### PR TITLE
Merge script findings api call

### PR DESCRIPTION
**Pull request by sebasrevuelta**
Adding the option to get findings for the primary branch.
Once merged, we can delete the script: `get_findings_primary_branch.py`

### CODE REVIEW

File: utilities/api/python_client_semgrep_api.py [hunk 1]
```diff
@@ -63,16 +64,20 @@ def get_all_findings(projects, headers):
...
-def get_findings_per_project(deployment_slug, project, headers):
+def get_findings_per_project(deployment_slug, project, primary_branch, headers):
     """
     Gets all findings for a project, and writes them to a file.
     The file format is equivalent to what the API would return if it weren't paginated.
     """
-    project_findings = retrieve_paginated_data(f"{BASE_URL}/{deployment_slug}/findings?repos={project}&dedup=false", "findings", 3000, headers=headers)
+    findings_url = f"{BASE_URL}/{deployment_slug}/findings?repos={project}&dedup=true"
+    if USE_PRIMARY_BRANCH_PARAM:
+        findings_url = f"{findings_url}&ref={primary_branch}"
```
Comment (armchairlinguist) [inline utilities/api/python_client_semgrep_api.py:78]: I think in this case maybe dedup can be false, as well?

File: utilities/api/python_client_semgrep_api.py [hunk 2]
```diff
@@ -63,16 +64,20 @@ def get_all_findings(projects, headers):
     """
     for project in projects['projects']:
         project_name = project['name']
+        primary_branch = project['default_branch']
```
Comment (armchairlinguist) [inline utilities/api/python_client_semgrep_api.py:67]:
```suggestion
        primary_branch = project['primary_branch']
```
The dedicated `primary_branch` attribute is what we should be using (see [doc](https://semgrep.dev/api/v1/docs/#tag/Project/operation/semgrep_app.saas.handlers.repository.openapi_list_recent_projects)). The default branch is just the SCM default and doesn't reflect if there have been overrides in Semgrep.

**Review by armchairlinguist — CHANGES_REQUESTED** [review]

**Comment by armchairlinguist** [comment]
Can you add the delete for the other file as part of this PR too? That makes more sense to me than doing it separately.

File: utilities/api/python_client_semgrep_api.py (hunk 1 above)
Comment (sebasrevuelta) [reply utilities/api/python_client_semgrep_api.py:78]: Yes!

File: utilities/api/python_client_semgrep_api.py (hunk 2 above)
Comment (sebasrevuelta) [reply utilities/api/python_client_semgrep_api.py]: Oh good catch!

**Review by armchairlinguist — APPROVED** [review]
Thanks for making those adjustments!

**Pull request merged**